docs/drvfc.rst

```diff
@@ -0,0 +1,152 @@
+=======================
+Firecracker driver
+=======================
+
+.. contents::
+
+Firecracker is an open source Virtual Machine Monitor (VMM) that
+runs on top of KVM. The focus of the project is on running VM's in a
+light-weight, secure way providing multi-tenant support. The features
+supported are minimalist to ensure the performance and security standards.
+
+The libvirt Firecrakcer driver is intended to be run as a system
+driver. The firecracker binary itself should be
+available in the path.
+
+Expected connection URI would be
+
+``fc:///system``
+
+
+Example guest domain XML configurations
+=======================================
+
+The Firecracker driver in libvirt is in its early stage of development
+supporting a subset of the features provided by the Firecracker VMM, for now being
+more a proof of concept
+
+Firecrackeer binaries can be found on the
+`github repository <https://github.com/firecracker-microvm/firecracker>`__
+
+Example of XML configuration
+
+::
+
+  <domain type='firecracker'>
+    <name>demo_domain</name>
+    <memory unit='GiB'>1</memory>
+    <vcpu>4</vcpu>
+    <os>
+      <type arch='x86_64' machine='pc'>hvm</type>
+      <kernel>/home/user/kernels/hello-vmlinux.bin</kernel>
+      <cmdline>panic=1</cmdline>
+      <root>vda</root>
+    </os>
+    <devices>
+      <disk type='file' device='disk'>
+        <source file='/home/ec2-user/LV/myownfiles/isos/hello-rootfs.ext4'/>
+        <target dev='vda' bus='virtio' />
+      </disk>
+
+      <serial type='pty'>
+        <target port='0'/>
+      </serial>
+
+      <interface type='ethernet'>
+        <target dev='tap0'/>
+        <guest dev='eth0'/>
+      </interface>
+
+      <interface type='ethernet'>
+        <target dev='tap2'/>
+        <guest dev='eth1'/>>
+      </interface>
+    </devices>
+  </domain>
+
+In the above example we specified the domain type to be `firecracker`,
+we specified the name of the domain, together with the memory and vcpus allocated.
+In the `<os>` tag we specified the source file for the kernel binary, the kernel command line.
+The only architecture supported for now is `x86_64` with the `hvm` type.
+The `<root>` tag specifies which of the following disk devices is used as the rootfs by matching the
+dev mount point attribute in the `<target>` tag.
+
```

Comment:
you may add a link to the docs about building a kernel and rootfs for FC: https://github.com/firecracker-microvm/firecracker/blob/main/docs/rootfs-and-kernel-setup.md